.github/workflows/compatibility.yml

```
@@ -0,0 +1,104 @@
+name: Test C compiler compatibility
+on:
+  push:
+    branches-ignore:
+      - devel
+      # Github Merge Queue temporary branches
+      - gh-readonly-queue/**
+
+  pull_request:
+    # Only consider PRs to devel
+    branches:
+      - devel
+
+    # Type of events to run CI on
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - ready_for_review
+
+  merge_group:
+    # Test all additions to the merge queue
+
+# Run every script action in bash
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  binaries:
+    name: Build binaries with most recent GCC version
+    runs-on: ubuntu-24.04
+    # Don't run for draft PRs
+    if: github.event_name != 'pull_request' || github.event.pull_request.draft == false
```

```suggestion
    if: ${{ !github.event.pull_request.draft }}
```

This one can be simplified a bit like this. Accessing operators yields `null` if a property is not found and accessing properties of `null` gives `null`, which is falsy.

We need the `${{}}` wrapping because `!text` is yaml tag syntax. Alternatively you could just use double quotes to wrap this.